subsys/usb/class/hid/core.c

```diff
@@ -349,19 +355,32 @@ static void hid_do_status_cb(struct hid_device_info *dev_data,
 #endif
 		break;
 	case USB_DC_CONNECTED:
-		LOG_DBG("USB device connected");
+		LOG_INF("Device connected");
 		break;
 	case USB_DC_CONFIGURED:
-		LOG_DBG("USB device configured");
+		LOG_INF("Device configured");
+		dev_data->configured = true;
 		break;
 	case USB_DC_DISCONNECTED:
-		LOG_DBG("USB device disconnected");
+		LOG_INF("Device disconnected");
+		dev_data->configured = false;
 		break;
 	case USB_DC_SUSPEND:
-		LOG_DBG("USB device suspended");
+		LOG_INF("Device suspended");
+		if (dev_data->configured) {
+			dev_data->configured = false;
+			dev_data->suspended = true;
+		}
 		break;
 	case USB_DC_RESUME:
-		LOG_DBG("USB device resumed");
+		LOG_INF("Device resumed");
+		if (dev_data->suspended) {
```

Comment:
I think we may only update the `suspended` flag on `USB_DC_SUSPEND`/`USB_DC_RESUME`.
We use both flags during check when we send report.